source/adios2/core/adios2_libinfo.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * adios2_libinfo.h
+ *
+ *  Created on: June 22, 2023
+ *      Author: Norbert Podhorszki pnorbert@ornl.gov
+ */
+
+#ifndef ADIOS2_LIBINFO_H_
+#define ADIOS2_LIBINFO_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern const int adios2_version_major;
+extern const int adios2_version_minor;
+extern const int adios2_version_patch;
+extern const char *const adios2_version_str;
+
+/** Return the list of available Engines in the installed adios2 library */
+void adios2_available_engines(int *nentries, char const *const **list);
```

Comment:
```suggestion
void adios2_available_engines(int *nentries, const char *const **list);
```
To match the style used in ADIOS2. AFAIK, const char and char const is the same. Same below